cpp/CMakeLists.txt

```diff
@@ -159,75 +126,149 @@ else()
-    "${CMAKE_CURRENT_SOURCE_DIR}/external/cub"
-    "${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}" 
-    "${CUDF_INCLUDE}"
-    "${NVGRAPH_INCLUDE}"
-    "${CMAKE_CURRENT_BINARY_DIR}/gunrock/"
-    "${CMAKE_CURRENT_BINARY_DIR}/gunrock/externals/moderngpu/include"
-    "${CMAKE_CURRENT_BINARY_DIR}/gunrock/externals/cub"
-    ${Boost_INCLUDE_DIRS}
+if(NVGRAPH_LIGHT MATCHES True)
+add_library(cugraph SHARED
+    src/utilities/grmat.cu
+    src/utilities/degree.cu
+    src/structure/cugraph.cu
+    src/link_analysis/pagerank.cu
+    src/traversal/bfs.cu
+    src/link_prediction/jaccard.cu
+    src/link_prediction/overlap.cu
+    src/converters/nvgraph.cu
+    src/converters/renumber.cu
+    src/community/nvgraph_gdf.cu
+    src/traversal/nvgraph_sssp.cu
+    src/traversal/two_hop_neighbors.cu
+    src/snmg/blas/spmv.cu
+    src/snmg/degree/degree.cu
+    src/snmg/utils.cu
+    src/nvgraph/arnoldi.cu
+    src/nvgraph/bfs.cu
+    src/nvgraph/bfs2d.cu
+    src/nvgraph/bfs_kernels.cu
+    src/nvgraph/convert.cu
+    src/nvgraph/csrmv.cu
+    src/nvgraph/csrmv_cub.cu
+    src/nvgraph/csr_graph.cpp
+    src/nvgraph/graph_extractor.cu
+    src/nvgraph/jaccard_gpu.cu
+    src/nvgraph/kmeans.cu
+    src/nvgraph/lanczos.cu
+    src/nvgraph/lobpcg.cu
+    src/nvgraph/matrix.cu
+    src/nvgraph/modularity_maximization.cu
+    src/nvgraph/nvgraph.cu
+    src/nvgraph/nvgraph_cusparse.cpp
+    src/nvgraph/nvgraph_cublas.cpp
+    src/nvgraph/nvgraph_error.cu
+    src/nvgraph/nvgraph_lapack.cu
+    src/nvgraph/nvgraph_vector_kernels.cu
+    src/nvgraph/pagerank.cu
+    src/nvgraph/pagerank_kernels.cu
+    src/nvgraph/partition.cu
+    src/nvgraph/size2_selector.cu
+    src/nvgraph/sssp.cu
+    src/nvgraph/triangles_counting.cpp
+    src/nvgraph/triangles_counting_kernels.cu
+    src/nvgraph/valued_csr_graph.cpp
+    src/nvgraph/widest_path.cu
+    ${CMAKE_CURRENT_BINARY_DIR}/gunrock/gunrock/util/test_utils.cu
+    ${CMAKE_CURRENT_BINARY_DIR}/gunrock/gunrock/util/error_utils.cu
+    ${CMAKE_CURRENT_BINARY_DIR}/gunrock/gunrock/util/misc_utils.cu
+    ${CMAKE_CURRENT_BINARY_DIR}/gunrock/externals/moderngpu/src/mgpucontext.cu
+    ${CMAKE_CURRENT_BINARY_DIR}/gunrock/externals/moderngpu/src/mgpuutil.cpp
+    ${CMAKE_CURRENT_BINARY_DIR}/gunrock/gunrock/util/gitsha1.c
 )
-
-###################################################################################################
-# - library paths ---------------------------------------------------------------------------------
-
-link_directories("${CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES}" # CMAKE_CUDA_IMPLICIT_LINK_DIRECTORIES is an undocumented/unsupported variable containing the link directories for nvcc
-                 "${CMAKE_BINARY_DIR}/lib"
-                 "${GTEST_LIBRARY_DIR}")
-
-###################################################################################################
-# - library targets -------------------------------------------------------------------------------
+else(NVGRAPH_LIGHT MATCHES True)
```

Comment:
We could get rid of this `else` branch I think

Reply:
Got it.